In [ ]:
!pip install googletrans==3.1.0a0

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.1/55.1 KB 6.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.4/133.4 KB 16.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.6/42.6 KB 4.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.5/1.5 MB 64.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 65.0/65.0 KB 8.1 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.6/53.6 KB 6.7 MB/s eta 0:00:00
  Created wheel for googletrans: filename=googletrans-3.1.0a0-py3-none-any.whl size=16368 sha256=84acfa7cdee2275b1a267cc89d20bb1e7da98a555a1f6c445c2ee5ed1aa02345
  Stored in directory: /root/.cache/pip/wheels/dd/59/af/8d6c96a719763990f1c548e36b17d9efdfb767f42f7ff39f53
Successfully built googletrans
  Attempting uninstall: chardet
    Found existing installation: chardet 4.0.0
 

In [ ]:
import numpy as np
import pandas as pd
from google.colab import files
from zipfile import ZipFile
import sys, os
import re
import time
from googletrans import Translator
import types
translator = Translator()

In [ ]:
data = pd.read_csv('data.csv',encoding='utf-8')

In [ ]:
data.head(5)

Page                                                Url  \
0     1  https://www.bucataras.ro/retete/mancare-de-lin...   
1     1  https://www.bucataras.ro/retete/mancare-de-ste...   
2     1  https://www.bucataras.ro/retete/fasole-boabe-c...   
3     1  https://www.bucataras.ro/retete/supa-crema-de-...   
4     1  https://www.bucataras.ro/retete/dovlecei-pane-...   

                                    Title  \
0                  Mancare de linte verde   
1                       Mancare de stevie   
2  Fasole boabe cu spanac in sos de rosii   
3        Supa crema de ciuperci Pleurotus   
4                   Dovlecei pane de post   

                                               Image  \
0                       https://www.bucataras.roNone   
1  https://www.bucataras.ro/uploads/modules/news/...   
2  https://www.bucataras.ro/uploads/modules/news/...   
3  https://www.bucataras.ro/uploads/modules/news/...   
4  https://www.bucataras.ro/uploads/modules/news/...   

                                             Content  \
0  O mâncare de post foarte gustoasă. Are un gust...   
1  Recunosc că încă nu m-am împrietenit cu ștevia...   
2                            Un pranz rapid de post.   
3  Supa crema de ciuperci Pleurotus este usoara s...   
4  Dovleceii pane sunt deliciosi chiar si in vari...   

                                         Ingredients Portion Cooking Time  \
0  150 g linte verde,\n3 linguri ulei,\nFrunzele ...       4           30   
1  Frunze de ștevie curățate cca. 500 g,\nCeapă g...       4           30   
2  1 ceapa rosie\n1 conserva fasole alba\n125 gr ...       2           30   
3  250 g. ciuperci Peurotus\n1 ceapa\n1 morcov\n1...       6           90   
4  3 dovlecei\nsare si piper\n3 cuburi supa de le...      40           60   

                                        Recipe Yield  
0                                                NaN  
1                                                NaN  
2                                                NaN  
3                                                NaN  
4  1) Cuburile de legume pot fi inlocuite cu vege...

In [ ]:
null_title = data['Title'].isnull().sum()
null_image = data['Image'].isnull().sum()
null_content = data['Content'].isnull().sum()
null_ingredients = data['Ingredients'].isnull().sum()
null_portion = data['Portion'].isnull().sum()
null_cooking_time =data['Cooking Time'].isnull().sum()
null_recipe_yield = data['Recipe Yield'].isnull().sum()

In [ ]:
print({
    'null_title': null_title,
    'null_image': null_image,
    'null_content': null_content,
    'null_ingredients': null_ingredients,
    'null_portion': null_portion,
    'null_cooking_time': null_cooking_time,
    'null_recipe_yield': null_recipe_yield
})

{'null_title': 0, 'null_image': 0, 'null_content': 1028, 'null_ingredients': 1, 'null_portion': 0, 'null_cooking_time': 0, 'null_recipe_yield': 2568}


## Null Recipe Yield: 2568
## Null Recipe Content: 1025
## Null Ingredient: 1

In [ ]:
data[data['Ingredients'].isnull()]['Url'].values

array(['https://www.bucataras.ro/retete/piept-de-pui-in-aluat-de-bere-23306.html'],
      dtype=object)

In [ ]:
#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
title_row = []
content_row = []
ingredients_row = []
recipe_yield = []


def translate_data(row, col):
  translated_value = ''
  translation = translator.translate(
      row,
      src="ro",
      dest='hu'
    ) 
  translated_value = translation.text

  print(translated_value)
  if col == 'Title':
    title_row.append(translated_value)
  elif col == 'Content':
    content_row.append(translated_value)
  elif col == 'Ingredients':
    ingredients_row.append(translated_value)
  elif col == 'Recipe Yield':
    recipe_yield.append(translated_value)


content_table = data.drop(['Page','Url','Image','Portion','Cooking Time'],axis = 1 )
content_table.replace(
        {
            '[\u0370-\u03FF]':'-',
            '\r\n':'-',
              '\xa0': '-'
        },
        regex = True,
        inplace = True
    )
#content_table.loc[77]['Content']
#data['Ingredients']
for col in content_table.columns:
  for index, value in content_table[col].iteritems():
    translate_data(value,col)


In [ ]:
#len(title_row)
content_table = data[['Page','Url','Image','Portion','Cooking Time']].copy()
title_row_df = pd.DataFrame(title_row, columns=['Title'])
title_row_df
content_table['Title'] = title_row_df['Title']
content_table

In [ ]:
content_row = pd.DataFrame(content_row, columns=['Content'])
content_row
content_table['Content'] = content_row['Content']
content_table

In [ ]:
ingredients_row = pd.DataFrame(ingredients_row, columns=['Ingredients'])
ingredients_row
content_table['Ingredients'] = ingredients_row['Ingredients']
content_table

In [ ]:
recipe_yield = pd.DataFrame(recipe_yield, columns=['Recipe Yield'])
recipe_yield
content_table['Recipe Yield'] = recipe_yield['Recipe Yield']
content_table

In [ ]:
content_table.replace('nan','',inplace=True)
content_table

Page                                                Url  \
0        1  https://www.bucataras.ro/retete/mancare-de-lin...   
1        1  https://www.bucataras.ro/retete/mancare-de-ste...   
2        1  https://www.bucataras.ro/retete/fasole-boabe-c...   
3        1  https://www.bucataras.ro/retete/supa-crema-de-...   
4        1  https://www.bucataras.ro/retete/dovlecei-pane-...   
...    ...                                                ...   
3435    89  https://www.bucataras.ro/retete/piure-de-urzic...   
3436    89  https://www.bucataras.ro/retete/cartofi-cu-roz...   
3437    89  https://www.bucataras.ro/retete/mucenici-pentr...   
3438    89  https://www.bucataras.ro/retete/fasole-batuta-...   
3439    89  https://www.bucataras.ro/retete/ciuperci-la-ta...   

                                                  Image Portion Cooking Time  \
0                          https://www.bucataras.roNone       4           30   
1     https://www.bucataras.ro/uploads/modules/news/...       4           30   
2     https://www.bucataras.ro/uploads/modules/news/...       2           30   
3     https://www.bucataras.ro/uploads/modules/news/...       6           90   
4     https://www.bucataras.ro/uploads/modules/news/...      40           60   
...                                                 ...     ...          ...   
3435  https://www.bucataras.ro/uploads/modules/news/...       6           60   
3436  https://www.bucataras.ro/uploads/modules/news/...       6           60   
3437  https://www.bucataras.ro/uploads/modules/news/...      12           30   
3438  https://www.bucataras.ro/uploads/modules/news/...       4           30   
3439                       https://www.bucataras.roNone       3           60   

                                 Title  \
0                     Zöld lencse étel   
1                          Stevia étel   
2     Bab spenóttal paradicsomszószban   
3             Pleurotus gombakrémleves   
4                 Gyors kenyér sütőtök   
...                                ...   
3435          Csalánpüré mamaligutával   
3436   Burgonya rozmaringgal a sütőben   
3437             Mártírok március 9-re   
3438                       Felvert bab   
3439                    Gomba a tálcán   

                                                Content  \
0     Nagyon finom böjti étel. Zöldbab íze van, nekü...   
1     Bevallom, még nem barátkoztam eléggé a steviáv...   
2                                      Gyors böjt ebéd.   
3     A Pleurotus gombakrémleves könnyű és kellemes ...   
4     A rántott sütőtök még böjtölt változatban is f...   
...                                                 ...   
3435  Frissen szedett, tavasz végén vagy tél elején ...   
3436  Nagyon szeretem a friss rozmaring ízét, ezért ...   
3437  A 40 szevastiai szent vértanú keresztény ünnep...   
3438  Ha böjt, akkor babot eszünk, ha pedig babot, a...   
3439                                   Jó gyorsétterem.   

                                            Ingredients  \
0     150 g zöld lencse,\n3 evőkanál olaj,\nEgy csom...   
1     A megtisztított stevia levelek kb. 500 g,\n1 s...   
2     1 vöröshagyma\n1 doboz fehér bab\n125 gr bébis...   
3     250 g Pleurotus gomba\n1 hagyma\n1 sárgarépa\n...   
4     3 tök\nsó, bors\n3 házi zöldségleves kocka\n15...   
...                                                 ...   
3435  csalán, minél több :)-2 kanál liszt 2 evőkanál...   
3436  1 kg piros burgonya, egy fej fokhagyma, néhány...   
3437  500 g szárított mártír\n4 l víz\n400 g cukor\n...   
3438  - bab - mindenki amennyit megfelelőnek tart, é...   
3439  1/2 kg gomba--egy csokor kapor--egy csokor zöl...   

                                           Recipe Yield  
0                                                        
1                                                        
2                                                        
3                                                        
4     1) A zöldségkockák helyettesíthetők házi zöld

In [ ]:
content_table.to_csv('ro_trans_hu.csv',index=False)
files.download('ro_trans_hu.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>